resilience4j-core/src/test/java/io/github/resilience4j/core/functions/NanosecondTimerTest.java

```diff
@@ -0,0 +1,39 @@
+package io.github.resilience4j.core.functions;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+/**
+ * Class NanosecondTimerTest.
+ */
+public class NanosecondTimerTest {
+
+    @Test
+    public void shouldPassElapsedTimetoConsumer() {
+        NanosecondTimer timer = NanosecondTimer.now();
+        final long now = System.nanoTime();
```

Comment:
Is there a way to either mock `System.nanoTime()` for the test or (more cleanly) give the `NanosecondTimer` constructor an optional `java.time.Clock` parameter and pass in a mock here? That would give us cleaner and less brittle tests. (I hadn't thought of this in the earlier comments.)

Reply:
We should not introduce a new class like `NanosecondTimer `. We should avoid to create class per call just to measure the time. This avoids GC pressure.

We usually use a MockClock to simulate the time.

Please have a look at https://github.com/resilience4j/resilience4j/blob/master/resilience4j-circuitbreaker/src/test/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachineTest.java#L46-L55

Reply:
Sure, will remove `NanosecondTimer`